mlflow/models/evaluation/base.py

```diff
@@ -653,40 +642,64 @@ def _evaluate(
     log resulting metrics & artifacts to MLflow Tracking.
 
     :param model: A pyfunc model instance, or a URI referring to such a model.
 
+    :param data: One of the following: A numpy array or list of evaluation features, excluding
+                 labels. Or a Pandas DataFrame, or a spark DataFrame, containing evaluation
+                 features and labels. If `feature_names` argument not specified, all columns are
+                 regarded as feature columns, otherwise column names which match `feature_names`
+                 are regarded as feature columns.
+
+    :param targets: If `data` is also a numpy array or list, A numpy array or list of evaluation
+                    labels. If `data` is a DataFrame, the string name of a column from `data`
+                    that contains evaluation labels.
+
     :param model_type: A string describing the model type. The default evaluator
                        supports "regressor" and "classifier" as model types.
-    :param dataset: An instance of :py:class:`mlflow.models.evaluation.EvaluationDataset`
-                    containing features labels (optional) for model evaluation.
-    :param run_id: The ID of the MLflow Run to which to log results. If
-                   unspecified, behavior depends on the specified `evaluator`.
-                   When `run_id` is unspecified, the default evaluator logs
-                   results to the current active run, creating a new active run if
-                   one does not exist.
-    :param evaluators: The name of the evaluator to use for model evaluations, or
-                       a list of evaluator names. If unspecified, all evaluators
-                       capable of evaluating the specified model on the specified
-                       dataset are used. The default evaluator can be referred to
-                       by the name 'default'. If this argument is unspecified, then
-                       fetch all evaluators from the registry.
-    :param evaluator_config: A dictionary of additional configurations to supply
-                             to the evaluator. If multiple evaluators are
-                             specified, each configuration should be supplied as
-                             a nested dictionary whose key is the evaluator name.
-    :return: An :py:class:`mlflow.models.evaluation.EvaluationDataset` instance containing
+
+    :param dataset_name: (Optional) The name of the dataset, must not contain double quotes (“).
+                         the name is logged to the `mlflow.datasets` tag. If not specified, the
+                         dataset hash is used as the dataset name.
+
+    :param dataset_path: (Optional) The path to a serialized DataFrame (e.g. a delta table,
+                         parquet file), must not contain double quotes (“). If specified,
+                         the path is logged to the `mlflow.datasets` tag.
```

Comment:
The dataset path doesn't necessarily have to refer to a DataFrame. It could refer to a serialized list or numpy array too.

How about?:

```suggestion
    :param dataset_path: (Optional) The path where the data is stored. Must not contain double quotes (“).
                         If specified, the path is logged to the `mlflow.datasets` tag.
```